packages/main/src/SideNavigation.js

```diff
@@ -0,0 +1,230 @@
+		/**
+		 * Defines the items in the <code>ui5-side-navigation</code>.
+		 *
+		 * @public
+		 * @slot
+		 */
+		"default": {
+			propertyName: "items",
+			type: HTMLElement,
+		},
+
+		/**
+		 * Defines the fixed items in the bottom of the <code>ui5-side-navigation</code>.
+		 *
+		 * @public
+		 * @slot
+		 */
+		fixedItems: {
+			type: HTMLElement,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.main.SideNavigation.prototype */ {
+		/**
+		 * Fired when the selection has changed via user interaction
+		 *
+		 * @event sap.ui.webcomponents.main.SideNavigation#selection-change
+		 * @param {HTMLElement} item the clicked item.
+		 * @public
+		 */
+		"selection-change": {
+			item: {
+				type: HTMLElement,
+			},
+		},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ *
+ * <h3>Usage</h3>
+ *
+ * <code>ui5-side-navigation</code> is used as a standard menu in applications. In order to add menu items use <code>ui5-side-navigation-items</code>.
+ *
+ * For the <code>ui5-side-navigation</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/SideNavigation.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.SideNavigation
+ * @extends UI5Element
+ * @tagname ui5-side-navigation
+ * @since 1.0.0-rc.8
+ * @appenddocs SideNavigationItem
+ * @public
+ */
+class SideNavigation extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return SideNavigationCss;
+	}
+
+	static get template() {
+		return SideNavigationTemplate;
+	}
+
+	static get staticAreaTemplate() {
+		return SideNavigationItemPopoverContentTemplate;
+	}
+
+	static async onDefine() {
+		await Promise.all([
+			ResponsivePopover.define(),
+			List.define(),
+			StandardListItem.define(),
+		]);
+	}
+
+	onBeforeRendering() {
+		this.items.forEach(item => {
+			item._collapsed = this.collapsed;
+		});
+
+		this.fixedItems.forEach(item => {
+			item._collapsed = this.collapsed;
+		});
+	}
+
+	handleItemClick(event) {
+		const currentItems = Array.from(this.querySelectorAll("ui5-side-navigation-item"));
+
+		currentItems.forEach(item => {
+			item.selected = item === event.target;
+		});
+
+		this.fireSelectionChange(event);
+
+		if (this.collapsed) {
+			this.openPicker(event.target);
+			this._popoverContent = event.target._generatePopoverContent();
+		}
+	}
+
+	fireSelectionChange(event) {
+		const item = (event.detail && event.detail.item && event.detail.item.item) || event.target;
+		const isArrowClicked = event.detail && event.detail.isIconClicked;
+
+		this.fireEvent("selection-change", {
+			item,
+			isArrowClicked,
+		});
+	}
+
+	getMiddleFocusHelper() {
+		return this.getDomRef().querySelector(".ui5-sn-middle-focus-helper");
+	}
+
+	_focusOtherList() {
+		this.getMiddleFocusHelper().focus();
+	}
+
+	focusNext(event) {
+		const eventTarget = event.target;
```

Comment:
did you consider using a List with list items, or Tree, to avoid implementing this focus handling